app/controllers/course/assessment/submission_question/comments_controller.rb

```diff
@@ -17,6 +15,13 @@ def create
 
       raise ActiveRecord::Rollback unless @post.save && create_topic_subscription && update_topic_pending_status
       raise ActiveRecord::Rollback unless @submission_question.save
+
+      # TODO: Remove @discussion_topic.touch.
+      #
+      # This is placed here because it was found that although the updated_at flag
+      # was updated on the ActiveRecord object, this change was not persisted to
+      # database. As a result, it was possible that the topic was not marked as read.
+      @discussion_topic.touch && mark_topic_as_read
```

Comment:
Does this mean `touch: true` in https://github.com/Coursemology/coursemology2/blob/master/app/models/course/discussion/post.rb#L17 is not working? Or it works in other places?

Reply:
Funny thing is it worked for programming file annotations and video topics ><

Reply:
I suspect it might because `submission_question` has an `updated_at` column while both file annotation and video topic don't have. AR or active_record-act_as might fail to work in such case.